src/app/shared/validator.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, ValidationErrors } from '@angular/forms';
+// Make sure not to import the entire rxjs library!!!
+import { Observable } from 'rxjs/Observable';
+import { timer } from 'rxjs/observable/timer';
+import { fromPromise } from 'rxjs/observable/fromPromise';
+
+import { MongoQueries } from './mongoQueries';
+import { CouchService } from './couchdb.service';
+
+import { switchMap, map } from 'rxjs/operators';
+
+@Injectable()
+export class ValidatorService {
+
+constructor(private couchService: CouchService, private mongoQueries: MongoQueries) {}
+
+  public CheckService$(findType: string, db: string, field: string, value: any): Observable<boolean> {
+    if (findType === 'findOneDocument') {
```

Comment:
For uniqueness we will always use the `findOneDocument` query, so we do not need to use `findType` in this function